File: Baekjoon/[11047] 동전 0.cs
namespace Program
{
    class Program
    {
        static void Main(string[] args)
        {
            int[] input = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
            int[] num = new int[input[0]];
            int sum = 0;
            int cnt = 0;

            for (int i = 0; i < input[0]; i++)
                num[i] = int.Parse(Console.ReadLine());

            int j = num.Length - 1;
            while (sum < input[1] || j >= 0)
            {
                if (num[j] + sum > input[1])
                {
                    j--;
                    continue;
                }
                sum += num[j];
                cnt++;
            }
            Console.Write(cnt);
        }
    }
}

File: Baekjoon/[11047] 동전 0.cpp
#include <vector>
#include <iostream>
using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(nullptr);

	vector<int> vec;
	int cnt = 0;
	int sum = 0;

	int n, k;
	cin >> n >> k;

	for (int i = 0; i < n; ++i)
	{
		int input;
		cin >> input;
		vec.push_back(input);
	}

	for (int i = n - 1; i >= 0; --i)
	{
		while (vec[i] + sum <= k)
		{
			sum += vec[i];
			++cnt;
		}
	}

	cout << cnt;
}